### ISSUE TITLE
OrmLite silently throws an exception with Guid columns in the SQLite provider

### ISSUE THREAD

**Issue by dabide** [ServiceStack/Issues#383]
Since v4.0.50, `Guid` columns can be written, but not read anymore with the SQLite provider. Here's a gist: https://gist.github.com/dabide/2998f6751b1a10ca7049

Apparently, the bug was introduced in the commit `fa188dc7b240c597638e0a7d27e02421a81b5ab6` of ServiceStack.OrmLite.

**Comment by dabide** [comment]
Here's the exception:
```
System.FormatException: Guid should contain 32 digits with 4 dashes (xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx).
   at System.Guid.TryParseGuidWithNoStyle(String guidString, GuidResult& result)
   at System.Guid.TryParseGuid(String g, GuidStyles flags, GuidResult& result)
   at System.Guid..ctor(String g)
   at System.Data.SQLite.SQLiteDataReader.GetGuid(Int32 i)
   at ServiceStack.OrmLite.Sqlite.Converters.SqliteGuidConverter.GetValue(IDataReader reader, Int32 columnIndex, Object[] values) in C:\Development\Repositories\ServiceStack.OrmLite\src\ServiceStack.OrmLite.Sqlite\Converters\SqliteGuidConverter.cs:line 36
   at ServiceStack.OrmLite.OrmLiteWriteCommandExtensions.PopulateWithSqlReader[T](T objWithProperties, IOrmLiteDialectProvider dialectProvider, IDataReader reader, Tuple`3[] indexCache, Object[] values) in C:\Development\Repositories\ServiceStack.OrmLite\src\ServiceStack.OrmLite\OrmLiteWriteCommandExtensions.cs:line 313
```

**Comment by mythz** [comment]
This test is passing for me without any Exceptions. What Sqlite package on NuGet are you using?

**Comment by dabide** [comment]
The exception isn't thrown, as OrmLite just logs it and continues. I compiled this using System.Data.SQLite.Core 1.0.98.1, as referenced by the ServiceStack.OrmLite.Sqlite.Windows project.

**Comment by dabide** [comment]
I just tested again, with a clean, new project and using the NuGet packages in stead of the projects from the ServiceStack.OrmLite repo. The result is the same: 4.0.50 fails, 4.0.48 passes. (Had to use System.Data.SQLite.Core 1.0.99.0, as it strangely enough is the only version now available on NuGet).

**Comment by mythz** [comment]
This is a bug in the System.Data.SQLite.Core ADO.NET provider where calling `reader.GetValues()` changes the behavior of `reader.GetGuid()` which would return a Guid before calling it, and throw a Format Exception anytime after GetValues() or GetValue() is called. I've added an config option to skip this `GetValues()` optimization in this commit which Sqlite.Windows has enabled by default in:
https://github.com/ServiceStack/ServiceStack.OrmLite/commit/36624b4f71168e99041655c91941ed1cb002cb7b

Note: this doesn't happen in the other ADO.NET providers, not even in [ServiceStack.OrmLite.Sqlite.Mono](https://www.nuget.org/packages/ServiceStack.OrmLite.Sqlite.Mono), which you could switch to, to avoid this error.

 This change is available from v4.0.51 that's now [available on MyGet](https://github.com/ServiceStack/ServiceStack/wiki/MyGet).
